src/main/groovy/org/jetbrains/intellij/IntelliJPlugin.groovy

```diff
@@ -82,13 +82,25 @@ class IntelliJPlugin implements Plugin<Project> {
                 configureProjectAfterEvaluate(subproject, subprojectExtension)
             }
         }
-        configureIntellijDependency(project, extension)
-        configurePluginDependencies(project, extension)
+
+        Configuration idea = project.configurations.create('idea')
+        Configuration ideaPlugins = project.configurations.create('ideaPlugins')
+
+        if (JavaPlugin.hasProperty('COMPILE_ONLY_CONFIGURATION_NAME')) {
+            project.configurations.getByName(JavaPlugin.COMPILE_ONLY_CONFIGURATION_NAME).extendsFrom ideaPlugins
```

Comment:
Currently, you only extend `ideaPlugins`. You need to extend both `idea` and `ideaPlugins` or extend `idea` in `ideaPlugins`. I'm not entirely sure if it makes more sense to apply both separately or to extend `idea` in `ideaPlugins`...

To extend both you could do something like the following so you don't need to specify both for all 3 calls:

```groovy
Configuration[] ideaConfigs = [idea, ideaPlugins]

project.configurations.getByName(/*...*/).extendsFrom ideaConfigs
```